app/Services/Registrar.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace Rogue\Services;
+
+use DoSomething\Gateway\Northstar;
+use Rogue\Repositories\CacheRepository;
+
+class Registrar
+{
+    /**
+     * Create new Registrar instance.
+     */
+    public function __construct()
+    {
+        $this->northstar = gateway('northstar');
+        $this->cache = new CacheRepository;
+    }
+
+    /**
+     * Finds a single user in the Rogue/Northstar system. If the user is not found
+     * in the Cache, then we grab it directly from Northstar and store in cache.
+     *
+     * @param  string $id
+     * @return object $user Northstar user object
+     */
+    public function find($id)
+    {
+        $user = $this->cache->retrieve($id);
+
+        if (! $user) {
+            $user = $this->northstar->getUser('id', $id);
+
+            // @TODO - How long should we store users in Cache?
+            $this->cache->store($user->id, $user);
```

Comment:
I'm not sure what you mean? Like only store the user if they have updated information?